src/libcore/ptr.rs

```diff
@@ -2509,6 +2509,29 @@ pub fn eq<T: ?Sized>(a: *const T, b: *const T) -> bool {
     a == b
 }
 
+/// Hash the raw pointer address behind a reference, rather than the value
+/// it points to.
+///
+/// # Examples
+///
+/// ```
+/// use std::collections::hash_map::DefaultHasher;
+/// use std::hash::Hasher;
+/// use std::ptr;
+///
+/// let five = 5;
+/// let five_ref = &five;
+///
+/// let mut hasher = DefaultHasher::new();
+/// ptr::hash(five_ref, &mut hasher);
+/// println!("Hash is {:x}!", hasher.finish());
+/// ```
+#[stable(feature = "rust1", since = "1.0.0")] // FIXME: replace with ???
+pub fn hash<T, S: hash::Hasher>(a: &T, into: &mut S) {
```

Comment:
Nit: prefer a more descriptive name than `a` for the reference. =P

Reply:
`x`? 😉 (it's "a" because ptr::eq uses a and b).  Where did you Haskell roots go, @Centril?!

But, seriously, is there an established convention for naming a reference?  I'm honestly not sure what you'd prefer (`a_reference`?)

Reply:
The roots burnt up due to the excessive amount of hot curry ingested.

Maybe `hashee` (in the sense of caller/callee) or `to_hash` or `to_be_hashed`?

Reply:
Hmm, `target` maybe? (or `hash_target`)

Reply:
I associated that with the Hasher, but it could work. :)